src/DistributedWriteAheadLog/Record.cpp

```diff
@@ -47,9 +54,17 @@ RecordPtr Record::read(const char * data, size_t size)
     /// FIXME, more graceful version handling
     assert(Record::version(flags) == VERSION);
 
-    NativeBlockInputStream input{rb, 0};
+    std::unique_ptr<NativeBlockInputStream> input;
+    std::unique_ptr<CompressedReadBuffer> compressed_in;
+    if (Record::compression(flags))
```

Comment:
Let's do the calculation inline here instead of in `Record::compression`

Reply:
I think in Record class, it has version() to get version, similarly compression() should be add to keep consistant. it can really do inline this compression via keep in Record class.